src/lib/Form/Data/Search/SearchData.php

```diff
@@ -0,0 +1,95 @@
+<?php
+
+/**
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace EzSystems\EzPlatformAdminUi\Form\Data\Search;
+
+class SearchData
+{
+    /** @var int */
+    private $limit;
+
+    /** @var int */
+    private $page;
+
+    /** @var string */
+    private $query;
```

Comment:
I never quite understood the need of encapsulation of data structure properties just to write getters and setters getting and setting them directly. This make sense to me only if structure is supposed to be immutable.

But not requesting changes, just asking why this is better. The flame war about this started around 2003 and never ended :)

Reply:
Accessing properties directly looks kinda lame. `$something->getProperty()` looks sexier. 😂

Reply:
I think the only reason is that other classes are the same :(